server/node/src/main/java/org/kaaproject/kaa/server/admin/client/util/Utils.java

```diff
@@ -174,14 +176,7 @@ public static String millisecondsToDateTimeString(long millis) {
    * @return boolean 'true' if email address is valid
    */
   public static boolean validateEmail(String mail) {
-    boolean result = false;
-    if (mail != null && mail.length() != 0) {
-      if (mail.indexOf('@') != INCORRECT_IDX
-          && mail.indexOf('.') != INCORRECT_IDX) {
-        result = true;
-      }
-    }
-    return result;
+    return !(mail == null || mail.length() == 0) && emailPattern.exec(mail) != null;
```

Comment:
Would you like to apply De Morgan's rule?
`!(mail == null || mail.length() == 0)` -> `mail != null && mail.length != 0`